src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java

```diff
@@ -121,6 +123,11 @@ public void setAccessModifiers(AccessModifier... accessModifiers) {
     }
 
     @Override
+    protected void registerCustomTypes(ConvertUtilsBean cub) {
```

Comment:
I think it is better to ask Module for converters - `getTypeConverters` and do all proper registration internally in checkstyle. Check should not be aware of how tat will be used, Check only provide converter for his types. The same as we do with Check's config resources for cache.
Does it make sense ?

Reply:
@romani If I understood correctly, you want me to convert `void registerCustomTypes(ConvertUtilsBean)` to `Converter[] getTypeConverters()` and only do registration in `AutomaticBean`?
This is fine, but it will have to be a `HashMap` as `register` takes `Converter` as well as the class that is being converted. So the method will be `Map<Class, Converter> getTypeConverters()`.

Reply:
One last comment, should  we name the method `getPropertyTypeConverters` so it is clear this only for property beans?

Reply:
lets make it `getPropertyTypeConverters`
yes, it should be a map.

Reply:
Done.